ytmusicapi/mixins/browsing.py

```diff
@@ -75,41 +98,60 @@ def search(self, query: str, filter: str = None) -> List[Dict]:
                     param2 = 'RAAGAAgACgA'
                 params = param1 + param2 + param3
 
-            body['params'] = params
-
-        response = self._send_request(endpoint, body)
-
-        try:
-            # no results
-            if 'contents' not in response:
-                return search_results
+        type = filter[:-1] if filter else None
 
-            if 'tabbedSearchResultsRenderer' in response['contents']:
-                results = response['contents']['tabbedSearchResultsRenderer']['tabs'][int(
-                    filter == "uploads")]['tabRenderer']['content']
-            else:
-                results = response['contents']
+        if not continuationToken:
+            body = {"query": q, "params": params}
 
-            results = nav(results, SECTION_LIST)
+        if continuationToken:
+            query = {"cToken": continuationToken, "continuation": continuationToken}
 
-            # no results
-            if len(results) == 1 and 'itemSectionRenderer' in results:
-                return search_results
+        response = self._send_request(endpoint, body, query=query)
 
-            for res in results:
-                if 'musicShelfRenderer' in res:
-                    results = res['musicShelfRenderer']['contents']
+        try:
+            results = None
+
+            if "continuationContents" in response:
+                results = response["continuationContents"]
+            elif "contents" in response:
+                if "tabbedSearchResultsRenderer" in response["contents"]:
```

Comment:
Is this part, dealing with `"tabbedSearchResultsRenderer"` correct?  I've never seen a response that includes `"tabbedSearchResultsRenderer"` so I have no idea if it works.

Reply:
It happens during upload search. Please upload at least a test song on your account so you can verify that the upload search works

Reply:
Thanks - I thought it might be something like that.  I'll try, but I haven't had any success getting the tests to work.  No doubt some misunderstanding/misconfiguration at my end.